camel/bots/discord_bot.py

```diff
@@ -0,0 +1,102 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import os
+from typing import TYPE_CHECKING, List, Optional
+
+from camel.agents import ChatAgent
+from camel.messages import BaseMessage
+
+# Conditionally import telebot types only for type checking
+if TYPE_CHECKING:
+    from discord import Message  # type: ignore[import-untyped]
+
+
+class DiscordBot:
+    r"""Represents a Discord bot that is powered by an agent.
+
+    Attributes:
+        chat_agent (ChatAgent): Chat agent that will power the bot.
+        channel_ids (List[int], optional): The channel IDs that the bot will listen to.
+        discord_token (str, optional): The bot token.
+    """
+
+    def __init__(
+        self,
+        chat_agent: ChatAgent,
+        channel_ids: Optional[List[int]] = None,
+        discord_token: Optional[str] = None,
+    ) -> None:
+        self.chat_agent = chat_agent
+        self.token = discord_token or os.getenv('DISCORD_TOKEN')
```

Comment:
move check for `self.token` into __init__?

Reply:
Could you explain more? The code is already in the init function.

Reply:
in `def run(self) -> None:` we check the token, maybe better to move that to here